ipamd/datastore/data_store.go

```diff
@@ -338,31 +337,32 @@ func (ds *DataStore) getDeletableENI() *ENIIPPool {
 			continue
 		}
 
-		log.Debugf("FreeENI: found a deletable ENI %s", eni.ID)
+		log.Debugf("getDeletableENI: found a deletable ENI %s", eni.ID)
 		return eni
 	}
 	return nil
 }
 
-// GetENINeedsIP finds out the eni in datastore which failed to get secondary IP address
+// GetENINeedsIP finds an ENI in the datastore that needs more IP addresses allocated
 func (ds *DataStore) GetENINeedsIP(maxIPperENI int64, skipPrimary bool) *ENIIPPool {
 	for _, eni := range ds.eniIPPools {
 		if skipPrimary && eni.IsPrimary {
 			log.Debugf("Skip the primary ENI for need IP check")
 			continue
 		}
 		if int64(len(eni.IPv4Addresses)) < maxIPperENI {
-			log.Debugf("Found eni %s that have less IP address allocated: cur=%d, max=%d",
+			log.Debugf("Found ENI %s that has less than the maximum number of IP addresses allocated: cur=%d, max=%d",
 				eni.ID, len(eni.IPv4Addresses), maxIPperENI)
 			return eni
 		}
 	}
 	return nil
 }
 
-// FreeENI frees a deletable ENI.
-// It returns the name of the ENI which is deleted out the data store
-func (ds *DataStore) FreeENI() string {
+// RemoveUnusedENIFromDataStore removes a deletable ENI from the data store.
+// It returns the name of the ENI which has been removed from the data store and needs to be deleted,
+// or empty string if no ENI could be removed.
+func (ds *DataStore) RemoveUnusedENIFromDataStore() string {
```

Comment:
I was going to say this should just be simply`RemoveENIFromDataStore` but then I realized there is also a DeleteENI and they neither calls the other they just do really similar things.  So I guess this should be refactored in a future PR.

Reply:
Definitely. This rename was just so we don't mix up the two FreeENI methods.